go.mod

```diff
@@ -8,6 +8,7 @@ require (
 	github.com/HabanaAI/gohlml v1.16.0
 	github.com/NVIDIA/go-dcgm v0.0.0-20240118201113-3385e277e49f
 	github.com/NVIDIA/go-nvml v0.12.0-1
+	github.com/beevik/etree v1.4.0
```

Comment:
Only My habit to use this library. I'll try encoding/xml then !

Reply:
@dave-tucker The issue here is that we don't know in advance the struct of the XML. So no possible way to use Unmarshall because it maps XML element to a struct. 
Etree lets you create a document and then FindElement into it.
If you have a better idea, please tell me, I'm making a second version now.